test/http2_test/test_goaway.py

```diff
@@ -0,0 +1,48 @@
+import logging
+import time
+import http2_base_server
+
+class TestcaseGoaway(object):
+  """ 
+    This test does the following:
+      Process incoming request normally, i.e. send headers, data and trailers.
+      Then send a GOAWAY frame with the stream id of the processed request.
+      It assert that the next request is made on a different TCP connection.
```

Comment:
This is actually not an assertion but the comment on line 27 explains the logic.